src/docker/shared.rs

```diff
@@ -335,7 +335,7 @@ impl Directories {
         Ok(Directories {
             cargo,
             xargo,
-            target,
+            target: metadata.target_directory.clone(),
```

Comment:
ideally, we'd not have this here, but sometimes we use `Directories`, sometimes we use `CargoMetadata`, ideally we'd only use one `CargoMetadata`